.github/workflows/performance_metrics_cron.yml

```diff
@@ -98,7 +103,22 @@ jobs:
           echo "PULUMI_TRACING_TAG_PULUMI_VERSION=$(pulumi version)" >> $GITHUB_ENV
           echo "PULUMI_TRACING_DIR=$PWD/traces"                      >> $GITHUB_ENV
       - name: Running Performance Matrix
-        run: make test_templates
+        shell: bash
+        # Even if some Go tests fail, others may have succeded and
+        # produced perf data; continue to try to upload it.
+        continue-on-error: true
+        run: |
+          set -euo pipefail
+          cd tests && go test -v -json -count=1 -cover -timeout 6h -parallel ${{ env.TESTPARALLELISM }} . 2>&1 | tee /tmp/gotest.log | gotestfmt
+        env:
+          PULUMI_PYTHON_CMD: python
+          TESTPARALLELISM: 3
+      - name: Upload traces GHA artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: traces
+          path: traces
```

Comment:
This can help debug issues with the data

Reply:
This is super smart!